### PR TITLE
feat: refine mlog_dump output

### PR DESCRIPTION
**Pull request by neverchanje**
## What problem does this PR solve?

mlog_dump used to print two lines for a mutation, one line for mutation header, and the other for mutation content. Two lines make it inconvenient for data searching (grep/awk). So we combined the two into one line in this PR.

For most of our use cases of mlog_dump, it's used to search data in the specified time range, in order to find hotkeys, or find business-related information. However, the mutation timestamp is printed as long int like "timestamp=1582715060503806", not suitable for time search. So we format the timestamp into "YYYY-mm-dd hh:mm:ss.SSS". For example: "timestamp=2020-02-26 19:04:20.453".

Result:
```
mutation [22.29.210.2138042763]: gpid=22.29, ballot=210, decree=2138042763, timestamp=2020-02-26 19:04:20.453, last_committed_decree=2138042762, log_offset=407050443493, log_length=204, update_count=1 [CHECK_AND_SET] "796a743221c0985c469c0dab0e093688" : IF SORT_KEY(first-start:) CT_VALUE_NOT_EXIST THEN SET SORT_KEY(first-start:) => VALUE(1582715060453) [expire=0]
```

### CODE REVIEW

**Comment by acelyc111** [comment]
```
mutation [22.29.210.2138042763]: gpid=22.29, ballot=210, decree=2138042763, timestamp=2020-02-26 19:04:20.453, last_committed_decree=2138042762, log_offset=407050443493, log_length=204, update_count=1 [CHECK_AND_SET] "796a743221c0985c469c0dab0e093688" : IF SORT_KEY(first-start:) CT_VALUE_NOT_EXIST THEN SET SORT_KEY(first-start:) => VALUE(1582715060453) [expire=0]
```
Now that `gpid`, `ballot` and `decree` have been printed in `[22.29.210.2138042763]`, how about remove the following `gpid=22.29, ballot=210, decree=2138042763, `

File: src/dist/replication/tool_lib/mutation_log_tool.cpp [hunk 1]
```diff
@@ -24,17 +24,10 @@
...
- *
- * Revision history:
- *     xxxx-xx-xx, author, first version
- *     xxxx-xx-xx, author, fix bug about xxx
- */
-
 #include <dsn/dist/replication/mutation_log_tool.h>
-#include "../lib/mutation_log.h"
+#include <dsn/utility/time_utils.h>
+#include "dist/replication/lib/mutation_log.h"
```
Comment (acelyc111) [inline src/dist/replication/tool_lib/mutation_log_tool.cpp]: code style

File: src/dist/replication/tool_lib/mutation_log_tool.cpp [hunk 2]
```diff
@@ -48,18 +40,21 @@ bool mutation_log_tool::dump(
     mutation_log_ptr mlog = new mutation_log_shared(log_dir, 32, false);
     error_code err = mlog->open(
         [mlog, &output, callback](int log_length, mutation_ptr &mu) -> bool {
-            if (mlog->max_decree(mu->data.header.pid) == 0)
+            if (mlog->max_decree(mu->data.header.pid) == 0) {
                 mlog->set_valid_start_offset_on_open(mu->data.header.pid, 0);
+            }
+            char timestamp_buf[32];
```
Comment (levy5307) [inline src/dist/replication/tool_lib/mutation_log_tool.cpp]: 这个不需要32位这么长吧？

Comment (neverchanje) [reply src/dist/replication/tool_lib/mutation_log_tool.cpp]: 22 bytes actually. 32 is for memory alignment.

**Pull request merged**